x-pack/filebeat/input/internal/httplog/roundtripper.go

```diff
@@ -167,6 +167,15 @@ func (rt *LoggingRoundTripper) RoundTrip(req *http.Request) (*http.Response, err
 	return resp, err
 }
 
+// TxID returns the current transaction.id value. If rt is nil, the empty string is returned.
+func (rt *LoggingRoundTripper) TxID() string {
```

Comment:
Possible to add test cases for this func?

Reply:
What failure modes are you expecting that can't be proven safe by inspection?

Reply:
Not just about failure scenarios. There is a specific format that is being returned here, so just thought if it is covered in a test to guard against unsafe changes in future. But you can take a call here. This might not change often or at all.

Reply:
I think I'm OK with what we have; the two places that return a transaction ID are within 10 lines of each other and are very clearly related. I did think about making them tied, but didn't see a good way to do it. I have that now, so I'll make that change.